src/api/ledger/parse/orderbook-order.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const _ = require('lodash');
+const utils = require('./utils');
+const flags = utils.core.Remote.flags.offer;
+const parseAmount = require('./amount');
+
+function parseOrderbookOrder(order: Object): Object {
+  const direction = (order.Flags & flags.Sell) === 0 ? 'buy' : 'sell';
+  const takerGetsAmount = parseAmount(order.TakerGets);
+  const takerPaysAmount = parseAmount(order.TakerPays);
+  const quantity = (direction === 'buy') ? takerPaysAmount : takerGetsAmount;
+  const totalPrice = (direction === 'buy') ? takerGetsAmount : takerPaysAmount;
+
+  const specification = utils.removeUndefined({
+    direction: direction,
+    quantity: quantity,
+    totalPrice: totalPrice,
+    passive: ((order.Flags & flags.Passive) !== 0) || undefined
+  });
+  // "quality" is omitted intentionally as it corresponds to
```

Comment:
> really "totalPrice" is the only one that is actually correct

Incorrect. Only `quality` (when it doesn't need to be inverted)  is correct in the sense that that's the rate you'll pay when executing the order. (Offers get executed at the rate they were placed at. The price/quality is static and retrieved from the directory nodes.  "Implementation detail, nanu nanu" ? 

I could almost buy the argument that if you can't be correct in all cases there's no point in trying, but I think you may as well be as accurate as you can be (like shying away from inverted qualities) 

> Adding a "price" field is dangerous because that "price" and "totalPrice" appear on equal footing,

I didn't really understand that.

I don't really feel strongly about this, but It may be throwing away information that may be of interest to someone. 

Questions are:

What is this api going to be used for? 
Who is going to use it? 
What are they all going to need?
Are all the orders sorted by price as they were before? 
Is there expectations that they will be?

Also, before the asks/bids were quite clearly, and simply grouped via virtue of being the responses to two requests. I'm not a trader, but I imagine people looking at orders want to know order depth etc, in the sense of availability bands. (I can't tell whether there's a bug, or you're trying to provide some new semantics I don't really understand but it seems like not all orders returned by rippled make it into the `{asks: [], bids: []}` object returned by getOrderBook).

Meaning, I'm not quite sure where you're going at with these changes.